src/workflow_tasks.rs

```diff
@@ -176,10 +184,14 @@ impl WorkflowTaskManager {
     /// Evict a workflow from the cache by its run id and enqueue a pending activation to evict the
     /// workflow. Any existing pending activations will be destroyed, and any outstanding
     /// activations invalidated.
+    /// In case if evict_from_cache flag is passed, run id will also be removed from the LRU cache.
     ///
     /// Returns that workflow's task info if it was present.
-    pub fn evict_run(&self, run_id: &str) -> Option<WorkflowTaskInfo> {
+    pub fn evict_run(&self, run_id: &str, evict_from_cache: bool) -> Option<WorkflowTaskInfo> {
```

Comment:
I don't love this extra flag. There's not any reason for it to be false except the line where you mention the lock is already held. We should just... not hold the lock there.

Reply:
Yeah, I hate it too, removed.